src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbVacancyReviewRepository.cs

```diff
@@ -14,47 +16,71 @@ namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Repositories
 {
     internal sealed class MongoDbVacancyReviewRepository : MongoDbCollectionBase, IVacancyReviewRepository
     {
+        private readonly ITimeProvider _timeProvider;
         private const string Database = "recruit";
         private const string Collection = "vacancyReviews";
 
-        public MongoDbVacancyReviewRepository(ILogger<MongoDbVacancyReviewRepository> logger, IOptions<MongoDbConnectionDetails> details) 
+        public MongoDbVacancyReviewRepository(
+            ILogger<MongoDbVacancyReviewRepository> logger, IOptions<MongoDbConnectionDetails> details, 
+            ITimeProvider timeProvider) 
             : base(logger, Database, Collection, details)
         {
+            _timeProvider = timeProvider;
         }
 
-        public async Task<List<VacancyReviewSearch>> SearchAsync(long vacancyReference)
+        public Task<List<QaVacancySummary>> SearchAsync(long vacancyReference)
```

Comment:
Should this summary class be generated in the orchestrator instead of the repo?

Reply:
I used projection instead of retrieving the whole document which also has vacancy snapshot now.